app/assets/javascripts/controllers/cloud_volume/cloud_volume_form_controller.js

```diff
@@ -26,6 +31,25 @@ ManageIQ.angular.app.controller('cloudVolumeFormController', ['$http', '$scope',
     miqService.sparkleOff();
   }
 
+  $scope.storageManagerChanged = function (id) {
+    miqService.sparkleOn();
+    
+    $http.get('/cloud_volume/storage_manager_form_fields/' + id)
+      .then(getStorageManagerFormDataComplete)
+      .catch(miqService.handleFailure);
+  }
+
+  function getStorageManagerFormDataComplete(response) {
+    var data = response.data;
+
+    $scope.afterGet = true;
+    $scope.cloudVolumeModel.emstype = data.emstype;
```

Comment:
Great. What about the next two lines where we just set the choices for the UI. I don't consider them being part of the model, but rather just supporting the UI, so I decided not to add them into model.